File: ContainerWithMostWater.java
public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            maxArea = Math.max(Math.min(height[left], height[right])* (right - left), maxArea);
            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }
        return maxArea;
    }
}

File: ContainerWithMostWater.py
class ContainerWithMostWater(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        left = 0
        right = len(height) - 1
        width = len(height) - 1
        maxArea = min(height[left], height[right]) * width
        
        while left < right:
            width-=1
            if height[left] < height[right]:
                left+=1
            else:
                right-=1
            maxArea = max(min(height[left], height[right]) * width, maxArea)
        return maxArea